qt/applications/workbench/workbench/plotting/toolbar.py

```diff
@@ -286,6 +286,8 @@ def _is_colorbar(cls, ax):
         return hasattr(ax, "_colorbar")
 
     def set_up_color_selector_toolbar_button(self, fig):
+        global current_ax_colour
```

Comment:
I am sure you already know this, but I will say it just in case I am wrong.

I used a global variable because Python treats variables as local by default within a function. Without global, modifying current_ax_colour inside the function throws an UnboundLocalError since it's trying to assign it locally before initialization. Declaring it as global lets me modify the global variable. An alternative could be testing it locally by doing current_ax_colour = None instead of global current_ax_colour to declare it at the start of the function but not globally, but for some reason that doesn't work.

I am not sure about what changed that caused this error. I am assuming it's because MPL 3.8 altered how it handles color collections so the return type might be different but I am not sure.

Reply:
There are no references in our code to `current_ax_colour`, apart from in that method, and previously it was a local variable. I think we should keep that variable local instead of changing global variables, and if that doesn't work then that needs investigating.

Reply:
I have just commited the fixes for this. :))